client/r/rhawc/R/rclientClass.R

```diff
@@ -0,0 +1,158 @@
+#' r client functions
+#'
+#' pretty please work
+#'
+#' @import httr
+#' @import plyr
+#' @export
+HawcClient <- function(baseUrl){
+
+  # local environment variables
+  thisEnv <- environment()
+
+  # persistent "state"
+  root_url <- baseUrl
+  #   token <- ""
+
+  checkIsAuthenticated <- function(){
+    if((nchar(get("token", thisEnv))==0)){
+      stop('Authentication required')
+    }
+    else if (get("token", thisEnv) == 'Unable to log in with provided credentials.') {
+      stop('Unable to log in with provided credentials.')
+    }
+  }
+  get_ = function(url) {
+    response = GET(url, add_headers(Authorization=paste('Token', get("token", thisEnv),sep=' '),
+                                    'Content-Type'="application/json"))
+    if (response$status_code >= 400 & response$status_code < 500) {
+      stop("An exception occured in the HAWC client module") }
```

Comment:
can we print the response or something too before stopping? Perhaps as a `warning`? b/c often times there's an explanation to what caused the error in the response